### PR TITLE
feat(core, wallet): implement address recovery

### LINKED ISSUE #219: [FEATURE]:  Implement Address Recovery
# Implement PIP-41: Address Recovery Algorithm

## Overview

Implement the address recovery algorithm as specified in [PIP-41](https://pips.pactus.org/PIPs/pip-41) to automatically discover and recover non-empty addresses when restoring a wallet from a mnemonic phrase. This implementation should follow the BIP-44 account discovery pattern similar to Bitcoin wallets.

## Background

Currently, when users restore a wallet from a mnemonic phrase, they need to manually recreate all their addresses. PIP-41 introduces an algorithm to automatically discover addresses that have been used (have balances or transaction history) during wallet restoration.

## Requirements

TBD

### Corner Case Test Scenario

**Original Wallet State:**
- Address 1 – Balance 1
- Address 2 – Balance 0 – No activity
- Address 3 – Balance 2
- Address 4 – Balance 0 – With activity (transaction history)
- Address 5 – Balance 0 – No activity

**Expected Recovered Wallet State:**
- Address 1 – Balance 1 ✅
- Address 2 – Balance 0 – No activity ❌ (should be skipped)
- Address 3 – Balance 2 ✅
- Address 4 – Balance 0 – With activity ✅
- Address 5 – Balance 0 – No activity ❌ (should be skipped)

### Additional Test Cases

1. **Empty Wallet**: Wallet with no addresses should return empty discovery
2. **Gap Limit Edge Cases**: Test with various gap limit values (1, 10, 50, 100)
3. **Network Failures**: Test behavior when RPC calls fail
4. **Large Address Sets**: Test with wallets having many addresses
5. **Mixed Usage Patterns**: Test wallets with scattered address usage

## Acceptance Criteria

- [ ] Address discovery algorithm implemented according to PIP-41
- [ ] Integration with existing wallet restoration process
- [ ] Proper error handling and retry logic for RPC calls
- [ ] Configuration options for gap limit and discovery parameters
- [ ] Comprehensive test coverage including corner cases
- [ ] Backward compatibility maintained
- [ ] Documentation updated
- [ ] Performance considerations for large address sets
- [ ] Network-specific behavior (mainnet/testnet)

## Dependencies

- Existing RPC client (`pactus-jsonrpc`)
- Trust Wallet Core for address derivation
- Current wallet storage and encryption systems

## References

- [PIP-41: Address Recovery](https://pips.pactus.org/PIPs/pip-41)
- [BIP-44: Account Discovery](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#user-content-Account_discovery)
- [BIP-42: Address Discovery](https://github.com/bitcoin/bips/blob/master/bip-0042.mediawiki) (to be added as reference)

## Priority

**High Priority** - This feature is essential for improving user experience during wallet recovery and aligns with industry standards for HD wallet implementations.

## Labels

- `enhancement`
- `wallet-core`
- `pip-41`
- `address-discovery`
- `high-priority`

### PR DESCRIPTION
**Pull request by phoenixit99**
Fixed #219

### CODE REVIEW

File: packages/wallet/src/wallet.ts [hunk 1]
```diff
@@ -310,6 +310,97 @@ export class Wallet {
...
+    };
+  }
+
+  /**
+   * Recover addresses according to PIP-41 specification
+   * @param password Password for wallet encryption
+   * @returns Promise<AddressInfo[]> - Array of recovered addresses
+   */
+  async recoverAddress(password: string): Promise<AddressInfo[]> {
+    const recoveredAddresses: AddressInfo[] = [];
```
[the remark this answers is not in the document]
Comment (phoenixit99) [reply packages/wallet/src/wallet.ts:362]: We don’t need to change PIP-41. This is a solution, but we can implement it in many ways, as long as the concept remains the same.

**Review by b00f — APPROVED** [review]
